packages/core/src/odata-common/request-builder/request-builder-base.ts

```diff
@@ -53,35 +53,72 @@ export abstract class MethodRequestBuilderBase<
    *
    * @param headers - Key-value pairs denoting additional custom headers
    * @returns The request builder itself, to facilitate method chaining
+   * @deprecated Since version 1.34.0 Use [[addCustomHeaders]] instead.
    */
   withCustomHeaders(headers: Record<string, string>): this {
     this.requestConfig.addCustomHeaders(headers);
     return this;
   }
 
+  /**
+   * Add custom headers to the request. If a header field with the given name already exists it is overwritten.
```

Comment:
Not sure, but somehow I stumble over this sentence.
```suggestion
   * Add custom headers to the request. Existing headers will be overwritten.
```